cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -189,15 +193,37 @@ protected Future<Pod> pod(Integer podId) {
     private Function<Pod, RestartReasons> podNeedsRestart;
 
     /**
-     * If allClient has not been initialized yet, does exactly that
+     * If brokerAdminClient has not been initialized yet, does exactly that
      * @return true if the creation of AC succeeded, false otherwise
      */
-    private boolean initAdminClient() {
-        if (this.allClient == null) {
+    private boolean initBrokerAdminClient() {
+        if (this.brokerAdminClient == null) {
             try {
-                this.allClient = adminClient(nodes, false);
+                this.brokerAdminClient = adminClient(nodes.stream().filter(NodeRef::broker).collect(Collectors.toSet()), false);
             } catch (ForceableProblem | FatalProblem e) {
-                LOGGER.warnCr(reconciliation, "Failed to create adminClient.", e);
+                LOGGER.warnCr(reconciliation, "Failed to create brokerAdminClient.", e);
+                return false;
+            }
+        }
+        return true;
```

Comment:
Is this necessary? As you said, IDE/compiler enforces to have a return statement outside the if statements. I don't see this if statements growing, since it's just a null check for one object. Otherwise we will have:
```
...
} else {
   return true;
}
return true;
```

Reply:
You don't need:
```
...
} else {
   return true;
}
return true;
```

You can just do 

```
...
} else {
   return true;
}
```

Reply:
I guess in this case the if part does not always return anyway, so I guess lets keep it for now, we can fix it later.